From 44f4d005175486b1e2b80f904786f245b999e490 Mon Sep 17 00:00:00 2001
From: Eugen Hristev <eugen.hristev@collabora.com>
Date: Mon, 27 Feb 2023 12:24:14 +0200
Subject: [PATCH 02/28] spl: spl_fit: add weak prototype for fpga_load

In case OPTIMIZE_DEBUG is set, unused code will not be optimized out, hence
the reference to fpga_load will be compiled.
if DM_FPGA and SPL_FPGA are not set, the build will fail with :

aarch64-none-linux-gnu-ld.bfd: common/spl/spl_fit.o: in function `spl_fit_upload_fpga':
u-boot/common/spl/spl_fit.c:595: undefined reference to `fpga_load'

By adding a weak prototype, build is successful.

Signed-off-by: Eugen Hristev <eugen.hristev@collabora.com>
---
 common/spl/spl_fit.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/common/spl/spl_fit.c b/common/spl/spl_fit.c
index 730639f756..0f0a9129fc 100644
--- a/common/spl/spl_fit.c
+++ b/common/spl/spl_fit.c
@@ -566,6 +566,12 @@ static void warn_deprecated(const char *msg)
 	printf("\tSee doc/uImage.FIT/source_file_format.txt\n");
 }
 
+__weak int fpga_load(int devnum, const void *buf, size_t bsize,
+		     bitstream_type bstype, int flags)
+{
+	return 0;
+}
+
 static int spl_fit_upload_fpga(struct spl_fit_info *ctx, int node,
 			       struct spl_image_info *fpga_image)
 {
-- 
2.39.2

